### PR TITLE
Implement `pthread_[sg]etname_np`

### PR DESCRIPTION
**Pull request by no92**
Blocked on managarm/managarm#522

### CODE REVIEW

File: options/posix/generic/pthread-stubs.cpp [hunk 1]
```diff
@@ -489,14 +489,30 @@ void pthread_cleanup_pop(int execute) {
 	frg::destruct(getAllocator(), hand);
 }
 
-int pthread_setname_np(pthread_t, const char *) {
-	mlibc::infoLogger() << "mlibc: pthread_setname_np is a stub" << frg::endlog;
+int pthread_setname_np(pthread_t thread, const char *name) {
+	if(strlen(name) > 15) {
```
Comment (ikbenlike) [inline options/posix/generic/pthread-stubs.cpp:493]: this check should take place in the sysdep

File: sysdeps/managarm/generic/fork-exec.cpp [hunk 2]
```diff
@@ -637,6 +639,62 @@ void sys_thread_exit() {
...
+	}
+
+	pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &cs);
+
+	int fd;
+	if(int e = sys_open(path, O_RDONLY | O_CLOEXEC, 0, &fd); e) {
+		return e;
+	}
+
+	if(int e = sys_read(fd, name, size, &real_size)) {
```
Comment (ikbenlike) [inline sysdeps/managarm/generic/fork-exec.cpp:686]: the same applies here as with the linux sysdep

File: sysdeps/linux/generic/sysdeps.cpp [hunk 3]
```diff
@@ -1117,6 +1119,62 @@ int sys_unlockpt(int fd) {
...
+	}
+
+	pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &cs);
+
+	int fd;
+	if(int e = sys_open(path, O_RDONLY | O_CLOEXEC, 0, &fd); e) {
+		return e;
+	}
+
+	if(int e = sys_read(fd, name, size, &real_size)) {
```
Comment (ikbenlike) [inline sysdeps/linux/generic/sysdeps.cpp:1166]: according to the manpage:

>        The pthread_getname_np() function can fail with the following
>       error:
>       ERANGE The buffer specified by name and len is too small to hold
>             the thread name.

I don't think `sys_read` performs any such check

**Review by ikbenlike — CHANGES_REQUESTED** [review]

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**